pkg/driver/node/mounter/systemd_mounter_test.go

```diff
@@ -387,7 +409,7 @@ sysfs /sys sysfs rw,seclabel,nosuid,nodev,noexec,relatime 0 0`),
 			err = os.WriteFile(procMountsPath, test.procMountsContent, 0755)
 			assertNoError(t, err)
 
-			mounter := &mounter.SystemdMounter{MountLister: &mounter.ProcMountLister{ProcMountPath: procMountsPath}}
+			mounter := &mounter.SystemdMounter{Mounter: mount.NewFakeMounter(parseProcMounts(test.procMountsContent))}
```

Comment:
I think we can change `test.procMountsContent` to be a `[]mount.MountPoint`, so we don't have to parse and convert them from a string. That why we wouldn't need `parseProcMounts` function.

Reply:
Thanks for reviewing, I was thinking about it at the first place, but I realised that for each test, the` procMountsContent` might be different, e.g: "non existing mount on /proc/mounts", so it might be cleaner to keep the parsing function in this case?

Reply:
Reviews have been addressed in the following commits.